wallet/wallet-cli-commands/src/command_handler/mod.rs

```diff
@@ -716,17 +718,43 @@ where
                 seconds_to_check_for_height,
                 check_all_timestamps_between_blocks,
             } => {
-                let timestamp_map = self
-                    .wallet()
-                    .await?
-                    .node_find_timestamps_for_staking(
+                let mut rng = make_true_rng();
+                let our_private_key = EndToEndPrivateKey::new_from_rng(&mut rng);
+                let our_public_key = our_private_key.public_key();
+
+                let wallet = self.wallet().await?;
+
+                let callee_public_key = wallet.e2e_public_key().await?.take();
+
+                let encrypted_input_data = wallet
+                    .get_timestamp_search_input_data(
+                        HexEncoded::new(our_public_key.clone()),
                         pool_id,
+                    )
+                    .await?;
+                let shared_secret = our_private_key.shared_secret(&callee_public_key);
+                let secret_input_data = shared_secret
+                    .decrypt_then_decode::<PoSTimestampSearchInputData>(&encrypted_input_data)?;
+
```

Comment:
Can't we pack all this in one function or something, instead of doing it here inline?